apis/v1alpha1/tcx_program_types.go

```diff
@@ -62,20 +62,24 @@ type TcxAttachInfoState struct {
 	AttachInfoStateCommon `json:",inline"`
 
 	// interfaceName is the Interface name to attach the tc program to.
-	InterfaceName string `json:"interfaceName"`
+	// +optional
+	InterfaceName string `json:"interfaceName,omitempty"`
 
 	// containerPid is the container pid to attach the tcx program in.
-	ContainerPid int32 `json:"containerPid"`
+	// +optional
+	ContainerPid int32 `json:"containerPid,omitempty"`
```

Comment:
This will be NetnsPath when my pr #411 goes in, but it's still required for the namespace scoped BpfApplication object.  It may not be required when interface discovery is enabled, so we'll need to figure out this validation too.